prf/utils/utils.py

```
@@ -362,63 +362,74 @@ def qs2dict(qs):
     return dictset(parse_qsl(qs))
 
 
-class TabRenderer(object):
-    def __init__(self, info):
-        pass
+def dict2tab(data, query_fields=None, headers=None, fmt='csv', auto=False):
+    import tablib
 
-    @classmethod
-    def dict2tab(cls, data, headers):
-        import tablib
+    query_fields = query_fields or []
+    headers = headers or []
 
-        def pop_(each, key):
-            val = each.pop(key, '')
-            if isinstance(val, (datetime, date)):
-                return val.strftime('%Y-%m-%dT%H:%M:%SZ')  # iso
-            else:
-                return unicode(val)
+    def pop_(each, key):
+        val = each.pop(key, '')
+        if isinstance(val, (datetime, date)):
+            return val.strftime('%Y-%m-%dT%H:%M:%SZ')  # iso
+        else:
+            return unicode(val)
+
+    if auto and data:
+        query_fields = data[0].flat(keep_lists=0).keys()
```

You might be overriding `query_fields`. I think you should raise an error if `auto` and `query_fields` are passed.